### ISSUE TITLE
Remove bloated and no longer required string.prototype.matchall dependency

### ISSUE THREAD

**Issue by ulrichstark** [surma/rollup-plugin-off-main-thread#58]
The `string.prototype.matchall` dependency to implement `String.prototype.matchAll()` in Node v10 was added because "The Workbox project [...] need to continue support Node v10 until our next major release version" (see #49).
`workbox-build` is now on version 7.0.0 and has a [minimum supported Node version of v16](https://github.com/GoogleChrome/workbox/blob/ee62b5b5b9ed321af457a2d962b2a34196a80263/packages/workbox-build/package.json#L15C5-L15C5).
`String.prototype.matchAll()` is available in Node since v12 [according to MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll#browser_compatibility).
The dependency tree of `string.prototype.matchall` is bloated, so i would like to have it removed as the polyfill is no longer required by the most recent `workbox-build` version.

Dependency tree of `string.prototype.matchall`:
![image](https://github.com/surma/rollup-plugin-off-main-thread/assets/8657779/28948800-ce5b-491c-98e4-ab45b7b8d5d1)

This screenshot is from an internal tool to analyze the `package-lock.json` file.
Dependencies starting from the second level below `string.prototype.matchall` are collapsed for better readability.
Removing no longer required dependency `string.prototype.matchall` would save users of this package from installing a lot of packages.

**Comment by donmccurdy** [comment]
I agree with @ulrichstark, and would be glad to trim the extra dependencies — I've opened a PR:

- https://github.com/surma/rollup-plugin-off-main-thread/pull/60

**Comment by ulrichstark** [comment]
I sadly don't think the PR will ever be merged. If you still depend on this package, consider just downgrading to version 2.2.2, which is the same as the current version 2.2.3 but without the `string.prototype.matchall` dependency looking at the commits.

I'm using the following in my `package.json`:
```json
{
    "// the following package is overriden because": "https://github.com/surma/rollup-plugin-off-main-thread/issues/58",
    "overrides": {
        "@surma/rollup-plugin-off-main-thread": "2.2.2"
    }
}
```
